core/lib/via_validation/src/utils.rs

```diff
@@ -0,0 +1,32 @@
+use std::{env, path::PathBuf};
+
+use crate::errors::VerificationError;
+
+/// Checks if the verification key exists for a given protocol version.
+pub async fn check_verification_key(protocol_version: u16) -> Result<(), VerificationError> {
```

Comment:
Just a suggestion, maybe we could make this function do all the checks and return the vk (like `load_verification_key`) and then we can use it inside `verify_snark` as well to avoid the code repetition?

Reply:
nice catch, I'll fix it